### ISSUE TITLE
EnforceCodeStyleInBuild does not detect unused usings without GenerateDocumentationFile

### ISSUE THREAD

**Issue by jnm2** [dotnet/roslyn#53720]
**Version Used**: dotnet 5.0.300

No errors or warnings are shown when running `dotnet build` like they are in the error list (or other variations on this, like having no TreatWarningsAsErrors or having editorconfig setting severity to error):

```cs
using System;
```

```ini
root = true

[*.cs]

# IDE0005: Using directive is unnecessary.
dotnet_diagnostic.IDE0005.severity = warning
```

```xml
<Project Sdk="Microsoft.NET.Sdk">

  <PropertyGroup>
    <TargetFramework>net5.0</TargetFramework>
    <EnforceCodeStyleInBuild>true</EnforceCodeStyleInBuild>
    <TreatWarningsAsErrors>true</TreatWarningsAsErrors>
  </PropertyGroup>

</Project>
```

Adding `<GenerateDocumentationFile>true</GenerateDocumentationFile>` fixes the problem, but I don't think I should have to add this. The point of EnforceCodeStyleInBuild is so that CI fails on things we miss that would bring clutter to the error list. If IDE0005 is showing in the error list without GenerateDocumentationFile, it should fail CI without GenerateDocumentationFile.

Besides that, it has awful discoverability to have to set GenerateDocumentationFile to get EnforceCodeStyleInBuild to work with IDE0005. GenerateDocumentationFile was a lucky guess based on a comment I saw on Twitter last month, and I'd already tried this four or five times since EnforceCodeStyleInBuild shipped and I completely failed and gave up each time.

/cc @jmarolf

**Comment by mavasani** [comment]
This is a duplicate of https://github.com/dotnet/roslyn/issues/41640

**Comment by jnm2** [comment]
Should I close or does that mess up the millstone?

**Comment by mavasani** [comment]
I think it is good to close out, let me do so.
